src/StreamingZipReader/StreamingZipReader.cs

```diff
@@ -108,10 +112,22 @@ public async ValueTask<bool> MoveToNextEntryAsync(bool skipDirectories, Cancella
         return true;
     }
 
+    private static void ParseZIP64ExtraField(ReadOnlySpan<byte> span, out long compressedSize, out long uncompressedSize)
+    {
+        byte[] tag = [1, 0];
+        var tagIndex = span.IndexOf(tag);
+        if (tagIndex == -1)
+            throw new InvalidDataException("ZIP64 file without extra field not supported.");
+        var reader = new SpanReader(span[tagIndex..]);
+        reader.Skip(4);
+        uncompressedSize = (long) reader.ReadUInt64LittleEndian();
+        compressedSize = (long) reader.ReadUInt64LittleEndian();
```

Comment:
What happens if you use `ulong` the whole way through instead of `long`?

Reply:
Opted for long since it has to become that anyway for the Stream interface. Not sure why they went for long instead of ulong in the Stream interface though...

Reply:
Gotcha, that makes sense.

CLS compliance (ulong is not a CLS-compliant type). They've dropped this rule in .NET API design for a while now though, if I remember right, but it was in effect for Stream.